src/include/network/network_defs.h

```diff
@@ -16,6 +16,9 @@ class ReadBuffer;
 // want anyone using it to directly access the socket downstream
 STRONG_TYPEDEF_HEADER(connection_id_t, uint16_t);
 
+// For threads
+#define CONNECTION_THREAD_COUNT 4
```

Comment:
It is not clear to me what this is doing, can you elaborate on the comment?

Reply:
We already have connection_thread_count being passed in to TerrierServer, why does RegisterProtocol need its own count? Are they related?